search/test.py

```diff
@@ -0,0 +1,76 @@
+    def setUp(self):
+        self.client = app.test_client()
+
+    def tearDown(self):
+        User.objects.delete()
+        Article.objects.delete()
+
+    def test_search_retrieve_several_user(self):
+        User(email='doejohn@test.com',
+             username='doejohn').set_password('search').save()
+        User(email='johndoe@test.com',
+             username='johndoe').set_password('search').save()
+        response = self.client.get('/search?q=john')
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(b'doejohn', response.data)
+        self.assertIn(b'johndoe', response.data)
+
+    def test_search_retrieve_several_article(self):
+        user = User(email='john@test.com',
+                    username='john').set_password('search').save()
+        article = Article(title='<h1>My fries are cold<br></h1>',
+                          content='Why my french fries are cold?')
+        article.author = user
+        article.save()
+        article = Article(title='<h1>I like fries<br></h1>',
+                          content='I like fries')
+        article.author = user
+        article.save()
+        response = self.client.get('/search?q=fries')
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(b'My Fries Are Cold', response.data)
+        self.assertIn(b'I Like Fries', response.data)
+
+    def test_search_retrieve_article_and_user(self):
+        user = User(email='fries@test.com',
+                    username='friesuser').set_password('search').save()
+        article = Article(title='<h1>My fries are cold<br></h1>',
+                          content='Why my french fries are cold ?')
+        article.author = user
+        article.save()
+        response = self.client.get('/search?q=fries')
+        self.assertEqual(response.status_code, 200)
+        self.assertIn(b'My Fries Are Cold', response.data)
+        self.assertIn(b'friesuser', response.data)
+
+    def test_search_maximum_10_users_and_articles(self):
+        # Check that /search retrieve 10 users and 10
+        # articles among 11 users and articles created
+        user = User(email='john@test.com',
+                    username='john').set_password('search').save()
+        for i in range(0, 11):
+            User(email='fries{}@test.com'.format(i),
+                 username='friesuser{}'.format(i)).set_password(
+                        'search').save()
+            article = Article(title='<h1>Title</h1>',
+                              content='My fries are cold')
+            article.author = user
+            article.save()
+        response = self.client.get('/search?q=fries')
+        self.assertEqual(response.status_code, 200)
+        count = 0
+        for word in response.data.split():
+            if b'fries' in word:
+                count = count + 1
+        # /search?q=fries appear in response.data and
+        # href for users appear twice
+        self.assertEqual(count, 31)
```

Comment:
That's why we should not just search for "searchtest" for checking that the value exists, but search for the value in a context: `<h2 name=username>searchtest</h2>`.